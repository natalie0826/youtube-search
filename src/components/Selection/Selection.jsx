import React from 'react';
import PropTypes from 'prop-types';

import './Selection.css';

export const Selection = (props) => {
    Selection.propTypes = {
        items: PropTypes.oneOfType([
            PropTypes.arrayOf(PropTypes.string),
            PropTypes.arrayOf(PropTypes.number)
        ]).isRequired,
        active: PropTypes.oneOfType([
            PropTypes.string,
            PropTypes.number
        ]).isRequired,
        onItemChanged: PropTypes.func.isRequired,
        title: PropTypes.string.isRequired
    };

    return (
        <div className="selection">
            <span htmlFor="selection">{props.title}</span>
            <select value={props.active} onChange={props.onItemChanged} onBlur={props.onItemChanged}>
                {props.items.map(item => (
                    <option value={item} key={item}>
                        {item}
                    </option>
                ))}
            </select>
        </div>
    );
};
