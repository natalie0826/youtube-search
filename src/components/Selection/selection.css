.selection {
    display: flex;
    flex-direction: column;
    margin: 20px 0;
}
  
.selection label {
    font-size: 20px;
    margin-bottom: 10px;
    color: #555;
}

.selection select {
    width: 100%;
    box-sizing: border-box;
    -webkit-box-sizing: border-box;
    -moz-box-sizing: border-box;
    border:1px solid #BEBEBE;
    padding: 7px;
    margin:0px;
    transition: all 0.30s ease-in-out;
    -webkit-transition: all 0.30s ease-in-out;
    -moz-transition: all 0.30s ease-in-out;
    -ms-transition: all 0.30s ease-in-out;
    -o-transition: all 0.30s ease-in-out;
    outline: none;
}

.selection select:focus {
    -moz-box-shadow: 0 0 8px #88D5E9;
    -webkit-box-shadow: 0 0 8px #88D5E9;
    box-shadow: 0 0 8px #88D5E9;
    border: 1px solid #88D5E9;
    transition: all 0.30s ease-in-out;
    -webkit-transition: all 0.30s ease-in-out;
    -moz-transition: all 0.30s ease-in-out;
    -ms-transition: all 0.30s ease-in-out;
    -o-transition: all 0.30s ease-in-out;
}
