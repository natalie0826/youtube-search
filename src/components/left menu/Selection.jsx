import React from 'react';
import PropTypes from 'prop-types';

import '../../styles/selection.css';

export const Selection = (props) => {
    Selection.propTypes = {
        items: PropTypes.array.isRequired,
        onItemChanged: PropTypes.func.isRequired,
        title: PropTypes.string.isRequired
    };

    return (
        <div className="selection">
            <label>{props.title}</label>
            <select value="" onChange={props.onItemChanged}>
                {props.items.map(item => (
                    <option value={item} key={item}>{item}</option>
                ))}
            </select>
        </div>
    )
}
