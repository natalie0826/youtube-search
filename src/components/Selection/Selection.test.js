import React from 'react';
import sinon from 'sinon';
import { shallow } from 'enzyme';
import { Selection } from './Selection';

describe('Selection functional component', () => {
    const selectionProps = {
        items: ['one', 'two', 'three', 'four', 'five'],
        onItemChanged: jest.fn(),
        title: 'title'
    }

    const component = shallow(<Selection {...selectionProps} />);

    it('has select tag', () => {
        expect(component.find('select').length).toBe(1);
    });

    it('expects selectedValue to be equal to the props value', () => {
        expect(component.find('select').props().value).toBe(selectionProps.selectedValue);
    });

    it('count of options should be egual with options.length (where options is an array passed as a prop)', () => {
        expect(component.find('option').length).toBe(selectionProps.items.length);
    });

    it('calls props method changeSelection after changing an option', () => {
        const callbackChange = sinon.spy();
        component.setProps({onItemChanged: callbackChange});
        expect(callbackChange.called).toBeFalsy();
        component.find('select').props().onChange({ target: {value: 'two'}});
        expect(callbackChange.called).toBeTruthy();
    });
});