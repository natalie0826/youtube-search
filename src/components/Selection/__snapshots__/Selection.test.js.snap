// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selection functional component matches its snapshot 1`] = `
<div
  className="selection"
>
  <span
    className="selection-title"
    htmlFor="selection"
  >
    title
  </span>
  <select
    className="select"
    onBlur={[Function]}
    onChange={[Function]}
    value="one"
  >
    <option
      key="one"
      value="one"
    >
      one
    </option>
    <option
      key="two"
      value="two"
    >
      two
    </option>
    <option
      key="three"
      value="three"
    >
      three
    </option>
    <option
      key="four"
      value="four"
    >
      four
    </option>
    <option
      key="five"
      value="five"
    >
      five
    </option>
  </select>
</div>
`;
